src/storage.ts

```diff
@@ -7,24 +7,85 @@ import { createPool as MySQLCreatePool } from 'mysql2';
 import pg from 'pg';
 import Cursor from 'pg-cursor';
 
-const storageTypes = {
+enum StoreType {
+  DataStore = 1,
+  MessageStore,
+  EventLog,
+}
+
+const backendTypes = {
   LEVEL    : 'level',
   SQLITE   : 'sqlite',
   MYSQL    : 'mysql',
   POSTGRES : 'postgres'
 };
 
+export function getDWNConfig(config: Config): DwnConfig {
+  let dataStore: DataStore = getStore(config.dataStore, StoreType.DataStore);
+  let eventLog: EventLog = getStore(config.eventLog, StoreType.EventLog);
+  let messageStore: MessageStore = getStore(config.messageStore, StoreType.MessageStore);
+  return { eventLog, dataStore, messageStore };
+}
+
+function getLevelStore(storeURI: URL, storeType: StoreType) {
+  switch (storeType) {
+  case StoreType.DataStore:
+    return new DataStoreLevel({
+      blockstoreLocation: storeURI.host + storeURI.pathname + '/DATASTORE',
+    });
+  case StoreType.MessageStore:
+    return new MessageStoreLevel({
+      blockstoreLocation: storeURI.host + storeURI.pathname + '/DATASTORE',
+    });
+  case StoreType.EventLog:
+    return new EventLogLevel({
+      location: storeURI.host + storeURI.pathname + '/EVENTLOG'
+    });
+  default:
+    throw new Error('Unexpected level store type');
+  }
+}
+
+function getDBStore(db: Dialect, storeType: StoreType) {
+  switch (storeType) {
+  case StoreType.DataStore:
+    return new DataStoreSql(db);
+  case StoreType.MessageStore:
+    return new MessageStoreSql(db);
+  case StoreType.EventLog:
+    return new EventLogSql(db);
+  default:
+    throw new Error('Unexpected level store type');
+  }
+}
+
+const getStore = (storeString: string, storeType) => {
+  const storeURI = new URL(storeString);
+  let store;
+  switch (storeURI.protocol.slice(0, -1)) {
+  case backendTypes.LEVEL:
+    store = getLevelStore(storeURI, storeType);
+    break;
+  case backendTypes.SQLITE || backendTypes.MYSQL || backendTypes.POSTGRES:
```

Comment:
so minor thing, but it js/ts usually recommendation is to group cases instead of logical operator, mostly for easier extendability and future support.
SO here it'd be 

`
case backendTypes.SQLITE:
case backendTypes.MYSQL:
case backendTypes.POSTGRES:
  store = getDBStore(getDBFromURI(storeURI), storeType);
  break;
`